arm/compute/client.go

```diff
@@ -0,0 +1,2031 @@
+// Copyright (c) Microsoft and contributors.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Code generated by Microsoft (R) AutoRest Code Generator 0.11.0.0
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+
+package compute
+
+import (
+	"fmt"
+	"github.com/azure/go-autorest/autorest"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+const (
+	ApiVersion             = "2015-06-15"
+	DefaultBaseUri         = "https://management.azure.com"
+	DefaultPollingDuration = 10 * time.Minute
+)
+
+type ComputeManagementClient struct {
+	autorest.Client
+	BaseUri        string
+	SubscriptionId string
+}
+
+func New(subscriptionId string) *ComputeManagementClient {
+	return NewWithBaseUri(DefaultBaseUri, subscriptionId)
+}
+
+func NewWithBaseUri(baseUri string, subscriptionId string) *ComputeManagementClient {
+	client := &ComputeManagementClient{BaseUri: baseUri, SubscriptionId: subscriptionId}
+	client.PollingMode = autorest.PollUntilDuration
+	client.PollingDuration = DefaultPollingDuration
+	return client
+}
+
+////////////////////////////////////////////////////////////////////////////////
+//
+// AvailabilitySets Client
+//
+////////////////////////////////////////////////////////////////////////////////
+type AvailabilitySetsClient struct {
+	ComputeManagementClient
+}
+
+func NewAvailabilitySetsClient(subscriptionId string) *AvailabilitySetsClient {
+	return NewAvailabilitySetsClientWithBaseUri(DefaultBaseUri, subscriptionId)
+}
+
+func NewAvailabilitySetsClientWithBaseUri(baseUri string, subscriptionId string) *AvailabilitySetsClient {
+	return &AvailabilitySetsClient{*NewWithBaseUri(baseUri, subscriptionId)}
+}
+
+// Delete the operation to delete the availability set.;
+//
+// resourceGroupName is the name of the resource group. availabilitySetName is
+// the name of the availability set.
+func (client *AvailabilitySetsClient) Delete(resourceGroupName string, availabilitySetName string) (err error) {
+
+	req, err := client.NewDeleteRequest(resourceGroupName, availabilitySetName)
+	if err != nil {
+		return fmt.Errorf("compute: Failure preparing AvailabilitySetsClient.Delete request (%v)", err)
```

Comment:
@ahmetalpbalkan Not precisely. In the model, an http.Request is *prepared* for use. Creating one never fails.